src/main/java/com/mineboundteam/minebound/entity/RockProjectile.java

```diff
@@ -0,0 +1,129 @@
+ * Derived from {@link net.minecraft.world.entity.projectile.AbstractArrow}
+ */
+public class RockProjectile extends AbstractArrow {
+    private double damage;
+    private double originalX;
+    private double originalY;
+    private double originalZ;
+    private int maxDist;
+    private float damageRadius;
+    public RockProjectile(EntityType<? extends RockProjectile> pEntityType, Level pLevel) {
+        super(pEntityType, pLevel);
+        //Adds trailing particles
+        this.setCritArrow(true);
+        this.setSoundEvent(SoundEvents.STONE_BREAK);
+        //Overwrite damage that would have been applied in AbstractArrow
+        this.setBaseDamage(0.0D);
+    }
+
+    public RockProjectile(Level pLevel, LivingEntity shooter, double pX, double pY, double pZ, double damage, float damageRadius, int maxDistance) {
+        super(ROCK_PROJECTILE.get(), pX, pY, pZ, pLevel);
+        this.setOwner(shooter);
+
+        this.damage = damage;
+        this.originalX = pX;
+        this.originalY = pY;
+        this.originalZ = pZ;
+        this.maxDist = maxDistance;
+        this.damageRadius = damageRadius;
+        this.setSoundEvent(SoundEvents.STONE_BREAK);
+        //Overwrite damage that would have been applied in AbstractArrow
+        this.setBaseDamage(0.0D);
+    }
+
+    @Override
+    protected void onHitEntity(EntityHitResult pResult) {
+        super.onHitEntity(pResult);
+        if(!this.level.isClientSide){
+            Entity target = pResult.getEntity();
+            Entity player = this.getOwner();
+            target.hurt(DamageSource.thrown(this, player), (float) damage);
+            if (player instanceof LivingEntity p ) {
+                this.doEnchantDamageEffects(p, target);
+            }
+        }
+    }
+
+    @Override
+    protected void onHitBlock(BlockHitResult pResult) {
+        super.onHitBlock(pResult);
+        if (!this.level.isClientSide) {
+            Entity entity = this.getOwner();
+            if (!(entity instanceof Mob) || net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.level, this)) {
+                BlockPos blockpos = pResult.getBlockPos().relative(pResult.getDirection());
+                if (this.level.isEmptyBlock(blockpos)) {
+                    List<Entity> hitEntities = this.level.getEntities(this, this.getBoundingBox().inflate(damageRadius));
+                    for (Entity target : hitEntities) {
+                        if (target instanceof LivingEntity) {
+                            target.hurt(DamageSource.thrown(this, entity), (float) damage);
+                            if (entity instanceof LivingEntity p) {
+                                this.doEnchantDamageEffects(p, target);
+                            }
+                        }
+                    }
+                }
+            }
+            this.discard();
+        }
+    }
+
+    @Override
+    protected ItemStack getPickupItem() {
+        return null;
```

Comment:
Return `ItemStack.EMPTY` here instead of `null`. This projectile should never be picked up but having this return `null` will cause the game to crash in the event it somehow does.